models/classes/search/ResultSetFilter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\search;
+
+use oat\oatbox\service\ConfigurableService;
+
+class ResultSetFilter extends ConfigurableService
+{
+    public function filter(array $content, string $structure): array
+    {
+        $allowedKeys = $this->getResultSetMapper()->map($structure);
+
+        return array_filter($content, function ($key) use ($allowedKeys) {
+            return $key === 'id' || in_array($key, array_keys($allowedKeys), true);
+        }, ARRAY_FILTER_USE_KEY);
```

Comment:
```suggestion
        return array_filter(
            $content, 
            function ($key) use ($allowedKeys) {
                return $key === 'id' || in_array($key, array_keys($allowedKeys), true);
            }, 
            ARRAY_FILTER_USE_KEY
        );
```

Please, apply PSR-12.

When breaking lines to pass parameters to a function, every parameter should be in a new line.

Reply:
This part is OK in my opinion. :)
You can find the same "breaking lines" in [PSR-12](https://www.php-fig.org/psr/psr-12/#47-method-and-function-calls):
```
somefunction($foo, $bar, [
  // ...
], $baz);

$app->get('/hello/{name}', function ($name) use ($app) {
    return 'Hello ' . $app->escape($name);
});
```

Reply:
It is not ok, there are 2 arguments here. read it: 

```
Argument lists MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument per line
```

Reply:
Take a look on this example:
![image](https://user-images.githubusercontent.com/59471572/101330002-29515780-3883-11eb-9d0d-d8081a91cc23.png)

Reply:
@shpran Yes, this is "optional" for anonymous functions and arrays, but honestly do you think it makes sense to have such different approaches?  This contradicts the first sentence.

```
When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument per line
```

For me is worse to read like this. It takes time to understand what is parameter for the anonymous function and what is not.

Reply:
Can I resolve this comment or you suggest to apply some changes here?

Reply:
I think nobody has issues with this format of coding except me, so I am closing :)